### PR TITLE
Create new manifest folder structure based on request client type

### PR DESCRIPTION
**Pull request by chintanrp**
Allow new parameter in request payload - client
Based on the client type choose manifest with modified folder structure of manifests

### CODE REVIEW

File: pyServer/GuestFishWrapper.py [hunk 1]
```diff
@@ -40,6 +41,7 @@ def __init__(self, rootLogger, handler, storageUrl, outputDirName, operationId,
         self.modeMinorSkipTo = modeMinorSkipTo
         self.kpThread = kpThread
         self.runWithCredscan = runWithCredscan
+        self.clientType = clientType
         self.osType = "unknown"
```
Comment (waynekuo) [inline pyServer/GuestFishWrapper.py:45]: What is no clientType input parameter is given?

File: pyServer/GuestFishWrapper.py [hunk 2]
```diff
@@ -398,20 +400,28 @@ def execute(self, storageUrl):
...
+                        clientFolder = parentFolder + self.clientType.lower() + os.sep
+                        manifestFile = clientFolder + osType + os.sep + self.mode.lower()
                         if not os.path.isfile(manifestFile):
                             self.rootLogger.warning("Manifest file " + manifestFile + " could not be located.")
-                            if not os.path.isdir(parentFolder + osType):
+                            if not os.path.isdir(parentFolder + self.clientType.lower()):
+                                # bad clientType in request payload
+                                self.WriteToResultFile(operationOutFile, "No manifests exists for client '" + self.clientType + "', trying 'supportability' as client...")
+                                self.clientType = "supportability"
+                                clientFolder = parentFolder + self.clientType.lower() + os.sep
```
Comment (waynekuo) [inline pyServer/GuestFishWrapper.py:413]: This feels like it should be a constant, but I guess we don't really do constants in our Python code?

File: pyServer/GuestFishWrapper.py [hunk 3]
```diff
@@ -398,20 +400,28 @@ def execute(self, storageUrl):
...
                                 self.WriteToResultFile(operationOutFile, "No manifests for OS '{0}', trying '{1}'...".format(osType,guessed_os) )
-                                manifestFile = parentFolder + osType + os.sep + self.mode.lower()
+                                manifestFile = clientFolder + osType + os.sep + self.mode.lower()
                             if not os.path.isfile(manifestFile):
                                 # bad manifest, try normal
                                 self.WriteToResultFile(operationOutFile, "No manifest exists for " + osType.lower() + " '" + self.mode.lower() + "' mode data collection. Trying 'normal' manifest...")
-                                self.mode = "normal"
-                                manifestFile = parentFolder + osType + os.sep + self.mode.lower()  #this should work...
+                                self.mode = "diagnostic"
+                                manifestFile = clientFolder + osType + os.sep + self.mode.lower()  #this should work...
```
Comment (waynekuo) [inline pyServer/GuestFishWrapper.py:424]: This feels like it should be a constant too.

File: pyServer/GuestFishWrapper.py [hunk 4]
```diff
@@ -398,20 +400,28 @@ def execute(self, storageUrl):
...
                         parentFolder = "/etc/azdis/"
-                        manifestFile = parentFolder + osType + os.sep + self.mode.lower()
+                        clientFolder = parentFolder + self.clientType.lower() + os.sep
+                        manifestFile = clientFolder + osType + os.sep + self.mode.lower()
                         if not os.path.isfile(manifestFile):
                             self.rootLogger.warning("Manifest file " + manifestFile + " could not be located.")
-                            if not os.path.isdir(parentFolder + osType):
+                            if not os.path.isdir(parentFolder + self.clientType.lower()):
+                                # bad clientType in request payload
+                                self.WriteToResultFile(operationOutFile, "No manifests exists for client '" + self.clientType + "', trying 'supportability' as client...")
```
Comment (waynekuo) [inline pyServer/GuestFishWrapper.py:411]: >client [](http://example.com/codeflow?start=98&length=6)

client type

File: pyServer/GuestFishWrapper.py (hunk 2 above)
Comment (chintanrp) [reply pyServer/GuestFishWrapper.py:414]: @waynekuo - How about this? We can keep adding constants in the common class.

File: pyServer/GuestFishWrapper.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (chintanrp) [reply pyServer/GuestFishWrapper.py]: upated

Comment (chintanrp) [reply pyServer/GuestFishWrapper.py]: updated

**Review by waynekuo — APPROVED** [review]
:shipit:

**Pull request merged**